### ISSUE TITLE
DPO Training: Loss rapidly converges to 0 and accuracy to 1, but model evaluation shows no improvement

### ISSUE THREAD

**Issue by bingo-G** [zhyang2226/OPA-DPO#10]
When I train my DPO model, the loss very quickly drops to almost 0, and the accuracy simultaneously approaches 1.0 (or 100%). This happens much faster than expected.

When I evaluate the trained model, I find that its outputs or behavior have almost no change compared to the base model before DPO training.

Have you encountered a similar problem? Any advice would be greatly appreciated. Thank you

**Comment by zhyang2226** [comment]
Please refer to the discussion in our paper for details.

It appears that both the preferred and rejected responses in your dataset were not generated by the target model. In this setting, the training loss can rapidly collapse toward zero due to the implicit KL constraint.

You may consider using our OPA-DPO. It can mitigate the phenomenon you observed, although it cannot completely eliminate it when the rejected responses are off-policy. Further, try smaller learning rate will also help.
